optreset is a BSD extension which neither glibc nor llvm-libc has. Putting
optind back to zero is how the scan is restarted in both, so llvm-libc goes
the same way glibc does.

--- a/lldb/source/Host/common/OptionParser.cpp
+++ b/lldb/source/Host/common/OptionParser.cpp
@@ -18,7 +18,7 @@
 void OptionParser::Prepare(std::unique_lock<std::mutex> &lock) {
   static std::mutex g_mutex;
   lock = std::unique_lock<std::mutex>(g_mutex);
-#ifdef __GLIBC__
+#if defined(__GLIBC__) || defined(__LLVM_LIBC__)
   optind = 0;
 #else
   optreset = 1;
